src/elfpy/simulators.py

```diff
@@ -41,11 +43,11 @@ def __init__(self, config_file):
         seconds_in_a_day = 86400
         self.time_between_blocks = seconds_in_a_day / self.config.simulator.num_blocks_per_day
         self.run_trade_number = 0
-        self.start_time = None
-        self.init_share_price = None
+        self.start_time: datetime.datetime = datetime.datetime.now()
```

Comment:
I don't think this should be in init. `start_time` was meant to indicate the start time of the simulation (i.e. `run_simulation()` was invoked), not the time when the simulator class was instantiated. Also, the version that used the time_utils function included a time zone of GMT.

Reply:
yeah you are right again.  I was trying to avoid having a type of `datetime | None` as that makes downstream things more annoying, but I'll just fix the downstream issues.